### PR TITLE
Add E2E testing to Overflow / Minor OverflowItem Bug Fix

### PR DESCRIPTION
**Pull request by lawrencewin**
### Platforms Impacted
- [ ] iOS
- [ ] macOS
- [x] win32 (Office)
- [ ] windows
- [ ] android

### Description of changes

This PR adds automation to Overflow, and it includes some minor bug fixes to the component.

- Add Overflow spec, page object, and constants within the E2E test package.
- Create Overflow E2E test section on test page.
- Fix `OverflowItem` bug where the item's `ViewProps` and styles were incorrectly merged with its cloned child. 

### Verification

Automated tests pass locally.

### Pull request checklist

This PR has considered (when applicable):
- [x] Automated Tests
- [ ] Documentation and examples
- [ ] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

**Review by rurikoaraki — APPROVED** [review]

File: apps/E2E/src/Overflow/specs/Overflow.spec.win.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
+import OverflowPageObject from '../pages/OverflowPageObject';
+
+describe('TabList Testing Initialization', () => {
```
Comment (samuelfreiberg) [inline apps/E2E/src/Overflow/specs/Overflow.spec.win.ts:3]: Update 'TabList'

File: apps/E2E/src/Overflow/specs/Overflow.spec.win.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  it('Wait for app load', async () => {
+    expect(await OverflowPageObject.waitForInitialPageToDisplay()).toBeTrue();
+  });
+
+  it('Click and navigate to TabList test page', async () => {
```
Comment (samuelfreiberg) [inline apps/E2E/src/Overflow/specs/Overflow.spec.win.ts:8]: Here too
